tools/codegen/selective_build/operator.py

```diff
@@ -0,0 +1,16 @@
+from dataclasses import dataclass
+
+from tools.codegen.model import OperatorName
+
+@dataclass(frozen=True)
+class SelectiveBuildOperator():
+    # The name of the operator. This includes the aten::, etc... prefix
+    name: OperatorName
```

Comment:
I noticed that here, and in other places, you have to do some contortions to handle the lack of `aten::` prefixes in the codegen in general. We could probably eliminate these all by teaching tools.codegen about namespaces. The basic concept would be to augment BaseOperatorName with a namespace field, and then modify the parsing process for `native_functions.yaml` to default these namespaces to be `aten::` (btw, this would pave the way for situations where people define non-aten namespaced functions in `native_functions.yaml`; right now this is inexpressible in the data model). Most of the codegen code here assume that they are on operating on an `aten` namespaced function, so we'd just add appropriate asserts and then ignore the namespace component for various things we wanted to do.

Reply:
I would strongly prefer to avoid a situation where OperatorName may or may not have a namespace (depending on context). That way leads to a lot of bugs.

Reply:
There is also some extra punning you are doing here, where `OperatorName` may or may not have an overload name. That's also the route to a lot of bugs that the type system could have caught for you. Just have two separate dataclasses, one for V1, one for V2, future readers will thank you for it :)

Reply:
> There is also some extra punning you are doing here, where OperatorName may or may not have an overload name. That's also the route to a lot of bugs that the type system could have caught for you. Just have two separate dataclasses, one for V1, one for V2, future readers will thank you for it :)

This one may be a tad tricky since in v1 selective build, the overload name is stripped even if the operator has no overload, so it's hard to say if (say) "aten::add" refers to the base function (which happens to have no overload name) or if it refers to the part before the DOT in aten::add.int.

Reply:
> I would strongly prefer to avoid a situation where OperatorName may or may not have a namespace (depending on context). That way leads to a lot of bugs.

I agree - this situation where sometimes the namespace is present, and sometimes not is not the best. For selective build we need the namespace, whereas for gen.py codegen, it isn't needed. I'll think of a way to separate the abstraction used internally so that this doesn't end up getting conflated.

Reply:
> There is also some extra punning you are doing here, where OperatorName may or may not have an overload name. That's also the route to a lot of bugs that the type system could have caught for you. Just have two separate dataclasses, one for V1, one for V2, future readers will thank you for it :)

Based on feedback from @ljk53, I'm settling on having a single unified data model for both (and not splitting the 2 at that layer). This should address the concern you have raised here.

Reply:
> This one may be a tad tricky since in v1 selective build, the overload name is stripped even if the operator has no overload, so it's hard to say if (say) "aten::add" refers to the base function (which happens to have no overload name) or if it refers to the part before the DOT in aten::add.int.

Well, the point is that in V1, you know the input source never has overload names, so you should parse it as BaseOperatorName. In V2, there may be overload names, so you parse it as OperatorName. The semantics of the input source tell you what it should be.

> Based on feedback from @ljk53, I'm settling on having a single unified data model for both (and not splitting the 2 at that layer). This should address the concern you have raised here.

sgtm